README.md

```diff
@@ -1,26 +1,25 @@
-# Lisp flag libraries
+# Command-line Flag Parsing
 
-Utilities to parse command line flags.
-The parse-command-line utility offers parsing of command line flags that
-is close to google3 binary flag parsing.
+The `ace.flag:parse-command-line` parses command line flags as defined by the
+`ace.flag:define` macro. `ace.flag:print-help` formats a `--help` message for
+all defined flags.
 
-The parser supports flags defined through the FLAG:DEFINE macro.
-ace.flag:print-help is a utility that formats the flags registered with define-flag.
+Flags may set special variables, if this feature is enabled. For example,
+`--cl:print-base` and `--cl:*print-base*` refer to the `*print-base*` special
+variable. This behavior is controlled by `--lisp-global-flags`.  The user may
+also allow this by default by passing `:global-flags t` to the
+`ace.flag:parse-command-line` function.
 
-Flags may refer to special variables. E.g. --cl:print-base and --cl::*print-base* refer to
-the '*print-base*' special variable. This behavior is controlled by --lisp-global-flags.
-The user may also allow this by default by passing :global-flags t
-to the PARSE-COMMAND-LINE function.
+Flags whose type is defined or expands to 'boolean' are considered boolean
```

Comment:
"Flags whose type is defined ..." I had to read that three times before realizing that it means "defined as boolean or expands to the boolean".  Also, can we give the full list of boolean values? The whole paragraph could use a rewrite.  How about:

Boolean flags are treated specially. Prefixing boolean flags with `no` sets the flag values to `nil` and no other arguments are consumed (e.g. `--nofoo` means to set the `foo` flag to `nil`). Boolean flags that are not followed by value arguments (that is, arguments that are not prefixed with dashes) are set to `t` and no more argument are consumed.  Boolean flags also accept explicit values: `yes`, `no`, `true`,  `false`, `t`, `null`, and `nil`. These values are consumed during the parsing of the boolean flags.

Reply:
Done